Library/Homebrew/extend/os/linux/diagnostic.rb

```diff
@@ -139,6 +139,34 @@ def check_linuxbrew_bottle_domain
           e.g. by using homebrew instead).
         EOS
       end
+
+      def check_gcc_dependent_linkage
```

Comment:
This could be very slow. On the other hand, `brew doctor` isn't run all that often.

Reply:
@carlocab Would be good to benchmark how much slower it ends up being?

Reply:
@Homebrew/linux would appreciate help with testing and benchmarking. Using Docker on macOS is not fun 😅

Reply:
Ok, this may take me a bit, but the exercise I plan to do is to install 100 randomly selected GCC dependents:

```sh
brew ruby <<EOS | xargs brew install
require 'formula'
gcc_dependents = Formula.all.select { |f| f.deps.map(&:name).include? "gcc" }
puts gcc_dependents.sample(100)
EOS
```

and then test `brew doctor` against this branch and against `master`.

Reply:
Assuming I did this correctly, `brew doctor` benchmarks should be in the markdown summary on this page in about an hour or so: https://github.com/carlocab/workflow-test/actions/runs/2804210379

Reply:
```
System.IO.IOException: No space left on device : '/home/runner/runners/2.294.0/_diag/Worker_20220805-142208-utc.log'
```

Oops.